src/safeds_runner/server/_memoization_map.py

```diff
@@ -290,21 +295,47 @@ def _update_stats_on_miss(
         self._map_stats[function_name] = stats
 
 
-def _convert_list_to_tuple(values: list) -> tuple:
+def _make_serializable(value: Any) -> Any:
     """
-    Recursively convert a mutable list of values to an immutable tuple containing the same values, to make the values hashable.
+    Make a value serializable.
 
     Parameters
     ----------
-    values : list
-        Values that should be converted to a tuple
+    value:
+        Value to be converted.
 
     Returns
     -------
-    tuple
-        Converted list containing all the elements of the provided list
+    converted_value:
+        Converted value.
     """
-    return tuple(_convert_list_to_tuple(value) if isinstance(value, list) else value for value in values)
+    if isinstance(value, dict):
+        return tuple((_make_serializable(key), _make_serializable(value)) for key, value in value.items())
+    elif isinstance(value, list):
+        return tuple(_make_serializable(element) for element in value)
+    elif _is_lambda(value):
```

Comment:
Block lambdas would not be covered in this case, as the check excludes them explicitly (as they're functions in python, not lambdas)

Reply:
That should work